libraries/node-core-library/src/FileSystem.ts

```diff
@@ -269,16 +331,45 @@ export class FileSystem {
    */
   public static move(options: IFileSystemMoveOptions): void {
     options = {
-      overwrite: true,
-      ensureFolderExists: false,
+      ...MOVE_DEFAULT_OPTIONS,
       ...options
     };
 
-    if (options.ensureFolderExists) {
-      FileSystem.ensureFolder(pathUtilities.basename(options.sourcePath));
+    try {
+      fsx.moveSync(options.sourcePath, options.destinationPath, { overwrite: options.overwrite });
+    } catch (error) {
+      if (options.ensureFolderExists) {
+        FileSystem.throwIfIsNotExistError(error);
+        const folderPath: string = pathUtilities.dirname(options.destinationPath);
```

Comment:
I was puzzled at first by the name `pathUtilities`.  Maybe we should call it something like `nodeJsPath` or `nativePath` to make it more clear that this is the `require('path')` API.